test/test_connection_timeout.cpp

```diff
@@ -107,3 +107,59 @@ TEST(Core, ConnectionTimeout) {
     (void)srt_epoll_release(pollid);
     (void)srt_cleanup();
 }
+
+/**
+ * The test creates a socket and tries to connect to a localhost port 5555
+ * in a blocking mode. The srt_connect function is expected to return
+ * SRT_ERROR, and the error_code should be SRT_ENOSERVER, meaning a
+ * connection timeout.
+ * This test is a regression test for an issue described in PR #833.
+ * Under certain conditions m_bConnecting variable on a socket
+ * might not be reset to false after a connection attempt has failed.
+ * In that case any further call to srt_connect will return SRT_ECONNSOCK:
+ * Operation not supported: Cannot do this operation on a CONNECTED socket
+ *
+*/
+TEST(Core, BlockingConnectionTimeoutLoop)
+{
+    using namespace std;
+    ASSERT_EQ(srt_startup(), 0);
+
+    const SRTSOCKET client_sock = srt_create_socket();
+    ASSERT_GT(client_sock, 0);    // socket_id should be > 0
+
+    sockaddr_in sa;
+    memset(&sa, 0, sizeof sa);
+    sa.sin_family = AF_INET;
+    sa.sin_port = htons(5555);
```

Comment:
I'd suggest making the 5555 port surely not bound with an SRT listener and surely not used by any other service. Otherwise the test may be thwarted by a service occupying port 5555.

To achieve that, you may start with 5000 and probably go up with the number while trying to bind a UDP socket to this port. Once successful, you can use this port in the test, then release it. For UDP it doesn't matter if anyone is reading from the port into which packets are otherwise coming.

Reply:
@ethouris Added UDP port check before connection timeout tests.